capstone/src/main/java/com/step902020/capstone/IndividualController.java

```diff
@@ -179,6 +179,71 @@ public RedirectView deleteSavedOrganization(
     return calendarEvents;
   }
 
+  /**
+   * get the recommended events for an individual
+   * @param currentUser user that is currently logged in
+   * @param count the number of events to be returned
+   * @return list of Events
+   */
+  @GetMapping("get-recommended-events-individual")
+  public List<Event> recommendEvents(CurrentUser currentUser,
+                                @RequestParam("count") String count) {
+    Individual targetUser = this.individualRepository.findFirstByEmail(currentUser.getEmail());
+    List<Individual> users = this.individualRepository.findByUniversity(targetUser.getUniversity());
+
+    int num = (count.equals("All")) ? Integer.MAX_VALUE : Integer.parseInt(count);
+    // find recommended events from other users
+    List<Event> recommended = Recommender.recommend(targetUser, users, u -> u.getSavedEvents(), num);
+    // if the list of recommended events is shorter than the list we want, add in more events from general event pool
+    if (recommended.size() < num) {
+      List<Event> allEvents = this.eventRepository.findByUniversity(targetUser.getUniversity());
+      int i = 0;
+      int length = 0;
+      int targetSize = recommended.size();
+      while (i < allEvents.size() && length < num - targetSize) {
```

Comment:
I'd make a temporary variable `int numExtraEvents =  num - targetSize` for ease of reading. The compiler will inline the definition anyway.